### DISCUSSION TITLE
Is there a way to substitute/relabel integral variables?

### DISCUSSION THREAD

**Discussion by CharlieDerby** [sympy/sympy#25422] [Q&A] [answered]
As far as I can tell, when `.subs()` is called on an `Integral` object, it will not act on variables being integrated over (i.e. any 'x' for which there is a 'dx' in the expression). Is there an in-built way to do this neatly or will I need to write my own implementation?

example:

```
from sympy import *
x, y, a, b = symbols('x y a b')

# What I type
Integral(x, (x,a,b)).subs(x,y)

# What I want to get
>>> Integral(y, (y,a,b))

# What I get
>>> Integral(x, (x,a,b))

# This gives the right behaviour
Integral(x, (x,a,b)).subs(b,y)

>>> Integral(x, (x,a,y))
```

Intended use case:

I have some integrals multiplied together which use the same integration variables and I want to write a function to merge them into one multi-integral. To do this I need to relabel the variables in one of them.

Thanks

**Comment by oscarbenjamin** [comment, marked as answer]
You can use `xreplace`.

**Reply by CharlieDerby** [reply]
Thanks, upon reading about it, `xreplace` seems appropriate for my purposes.
